tidb-lightning/tidb-lightning-backends.md

```diff
@@ -0,0 +1,465 @@
+---
+title: TiDB Lightning Import Modes
+summary: Learn the different import modes of TiDB Lightning.
+aliases: ['/docs/dev/tidb-lightning/tidb-lightning-tidb-backend/','/docs/dev/reference/tools/tidb-lightning/tidb-backend/','/tidb/dev/tidb-lightning-tidb-backend']
+---
+
+# TiDB Lightning Import Modes
+
+TiDB Lightning supports three backends: Importer-backend (default), Local-backend, and TiDB-backend. It determines how `tidb-lightning` delivers data into the target cluster.
+
+The **Importer-backend** (default) requires `tidb-lightning` to first encode the SQL or CSV data into KV pairs, and relies on the external `tikv-importer` program to sort these KV pairs and ingest directly into the TiKV nodes.
+
+The **Local-backend**: `tidb-lightning` first encodes data into key-value pairs, sorts and stores them in a local temporary directory, and writes these key-value pairs to each TiKV node in batches. Then, TiKV ingests these key-value pairs into the cluster. The implementation of Local-backend is the same with that of Importer-backend but does not rely on the external `tikv-importer` component.
+
+The **TiDB-backend** requires `tidb-lightning` to encode these data into SQL `INSERT` statements, and has these statements executed directly on the TiDB node.
+
+| Back end | Local-backend | Importer-backend | TiDB-backend |
+|:---|:---|:---|:---|
+| Speed | Fast (~500 GB/hr) | Fast (~300 GB/hr) | Slow (~50 GB/hr) |
+| Resource usage | High | High | Low |
+| Network bandwidth usage| High | Medium | Low |
+| ACID respected while importing | No | No | Yes |
+| Target tables | Must be empty | Must be empty | Can be populated |
+| External component needed | No | `tikv-importer` | No |
+| TiDB versions supported | >= v4.0.0 | All | All |
+
+## How to choose the backend modes
+
+- If the target cluster of data import is v4.0 or later versions, consider using the Local-backend mode first, which is easier to use and has higher performance than that of the other two modes.
+- If the target cluster of data import is v3.x or earlier versions, it is recommended to use the Importer-backend mode.
+- If the target cluster of data import is in the online production environment, or if the target table of data import already has data on it, it is recommended to use the TiDB-backend mode.
+
+## TiDB Lightning Local-backend
+
+The Local-backend feature was introduced in TiDB v4.0.3. TiDB Lightning only of v4.0.3 or later versions has this feature. In addition, Local-backend only supports TiDB cluster of v4.0.0.
```

Comment:
```suggestion
The Local-backend feature was introduced in TiDB v4.0.3. TiDB Lightning only of v4.0.3 or later versions has this feature. In addition, Local-backend only supports TiDB clusters of v4.0.0 or above.
```
I'm confused because the first two sentences seem to conflict with the last sentence. So can a v4.0.1 TiDB cluster use the Local-backend feature?

Reply:
The original Chinese is `Local-backend 特性在 TiDB v4.0.3 发布，v4.0.3 及以上的 TiDB Lightning 才包含此特性。另外 Local-backend 只支持 v4.0.0 以上的集群。` @glorv Could you answer the question above?

Reply:
We release `local backend` in reversion v4.0.3. But it can support restore data to tidb clusters of v4.0.0 or above(include v4.0.0~4.0.2).
I want to express only with tidb-lightning >= v4.0.3  can we use this feature. But It's ok for all tidb clusters >= v4.0.0

Reply:
This conflict is caused by the fact that we always release tidb(tidb/tikv/pd ...) and its toolset (lightning/br...) together, but the toolset is often compatible with multi versions of tidb

Reply:
```suggestion
The Local-backend feature is introduced to TiDB Lightning since TiDB v4.0.3. But you can use this feature to import data to TiDB clusters of v4.0.0 or above.
```
How about saying in this way?

Reply:
And for Chinese: `自 TiDB 4.0.3 版本起，TiDB Lightning 引入了 Local-backend 特性。该特性支持导入数据到 v4.0.0 以上的 TiDB 集群。` @TomShawn @glorv